### ISSUE TITLE
Allow integers for PostgreSQL JSON operators

### ISSUE THREAD

**Issue by BlackWiCKED** [ScientaNL/DoctrineJsonFunctions#48]
Maybe there is a proper way to do this, but I did not find it. I want to select the first value of a JSON array, that is stored property in database like: ["VALUE"].

According to https://www.postgresql.org/docs/9.3/functions-json.html, the operator I should use is:

`["VALUE"]->>0`

For some reason I can't target the integer value when using JSON_GET_TEXT(jsondoc, path). If I go

```
$query->andWhere("JSON_GET_TEXT(i.value, 0) = :fieldValue)
      ->setParameter('fieldValue', 'VALUE');
```
I get the following error:

``Error: Expected StateFieldPathExpression | string | InputParameter | FunctionsReturningStrings | AggregateExpression, got '0'``

If I send the zero as a string, like this:

```
$query->andWhere("JSON_GET_TEXT(i.value, '0') = :fieldValue)
      ->setParameter('fieldValue', 'VALUE');
```
The genrated SQL will look like this: ``["VALUE"]->>'0'`` and it won't give the desired results.

I found a workaround, but there has to be a better way to do this. I wrap up the index number into ABS function like this:

```
$query->andWhere("JSON_GET_TEXT(i.value, ABS(0)) = :fieldValue)
      ->setParameter('fieldValue', 'VALUE');
```

**Comment by Hikariii** [comment]
You are right, the postgres functionNode needs to be refactored like the mysql one.
The stringPrimary must be followed by any value like this https://github.com/ScientaNL/DoctrineJsonFunctions/blob/master/src/Query/AST/Functions/Mysql/JsonArrayInsert.php#L16

**Comment by Hikariii** [comment]
Fixed and tested by https://github.com/ScientaNL/DoctrineJsonFunctions/pull/56/files

**Comment by Hikariii** [comment]
Released in 4.1.1 https://packagist.org/packages/scienta/doctrine-json-functions#4.1.1
